Gui/python/CustomizedWidget.py

```diff
@@ -90,63 +115,89 @@ def getVDDD(self, pChipID):
 
 class ChipBox(QWidget):
     chipchanged = pyqtSignal(int, int)
-
-    def __init__(self, master,pChipType,serialNumber):
+    #adding default value to serialNumber="RH0009" can prevent ChipBox from crashing under online mode
+    def __init__(self, master,pChipType,serialNumber="RH0009"):
         super().__init__()
         logger.debug("Inside ChipBox")
         self.master = master
         self.connection = self.master.connection
+        self.serialNumber=serialNumber
         self.chipType = pChipType
         self.mainLayout = QHBoxLayout()
-        self.ChipList = []
-        #       self.initList()
+        self.ChipList = [] #chip id list for a single module
+        #self.initList()
         self.createList()
         self.VDDAMap = {}
         self.VDDDMap = {}
         self.ChipGroupBoxDict = {}
-        self.serialNumber=serialNumber
+        
+        
         #get trim values from DB
         GetTRimFromDB=GetTrimClass()
-        #adding a get connection method to get the real time changes?
         GetTRimFromDB.connection=self.connection
+
+
         if GetTRimFromDB.connection == "Offline" or GetTRimFromDB.connection == []:
-            print("serial numer : " + str(self.serialNumber))
-            print(GetTRimFromDB.connection)
             for chipid in self.ChipList:
                 self.ChipGroupBoxDict[chipid] = self.makeChipBox(chipid)
 
+
         
         else:
             if self.serialNumber != []:
-                #need to find way to haddle serial number iussue, it is not [] and None if the chip box is empty, so how we handdle it?
-                sorted_VDDAlist,sorted_VDDDlist= GetTRimFromDB.GetTrim(self.serialNumber)
-            
+                
+                sorted_VDDAlist,sorted_VDDDlist= GetTRimFromDB.GetTrim(self.serialNumber)                
+
+                #case for can't find anything on the database
                 if sorted_VDDAlist == [] or sorted_VDDDlist == []:
-                    #print("we can't find the trim values for the module with serial number " + str(self.serialNumber))
-                    #print(GetTRimFromDB.connection)
                     for chipid in self.ChipList:
-                        self.ChipGroupBoxDict[chipid] = self.makeChipBox(chipid)
+                        VDDA = "can't find the value"
+                        VDDD = "can't find the value"
+                        self.ChipGroupBoxDict[chipid] = self.makeChipBoxWithDB(chipid,VDDA,VDDD)
                 
                 
-                else:
+
+                if sorted_VDDAlist != [] or sorted_VDDDlist != []:
                     i = 0
                     for chipid in self.ChipList:
```

Comment:
You can change this to use enumerate
for i, chipid in enumerate(self.ChipList):
 ...